### PR TITLE
[master] 3.3.5 -> master cherry-pick

### PR DESCRIPTION
**Pull request by funjoker**
**Changes proposed:**

-  Cherry picking

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

### CODE REVIEW

**Comment by illfated** [comment]
You probably know this already, but for those who haven't found out yet:
```cpp
/home/travis/build/TrinityCore/TrinityCore/src/server/game/AI/SmartScripts/SmartScript.cpp:301:129:
 fatal error: 
      too many arguments to function call, expected at most 2, have 3
  ...target->PlayDistanceSound(e.action.sound.sound, e.action.sound.onlySelf ? target->ToPlayer() : nullptr, e.action.sound.keyBroadcastTextId);
     ~~~~~~~~~~~~~~~~~~~~~~~~~                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/travis/build/TrinityCore/TrinityCore/src/server/game/Entities/Object/Object.h:474:9: note: 
      'PlayDistanceSound' declared here
        void PlayDistanceSound(uint32 soundId, Player* target = nullptr);
        ^
1 error generated.
make[2]: *** [src/server/game/CMakeFiles/game.dir/AI/SmartScripts/SmartScript.cpp.o] Error 1
```

**Comment by funjoker** [comment]
Ya, gonna solve that :D

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 1]
```diff
@@ -297,7 +297,11 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
             {
                 if (IsUnit(target))
                 {
-                    target->PlayDirectSound(e.action.sound.sound, e.action.sound.onlySelf ? target->ToPlayer() : nullptr, e.action.sound.keyBroadcastTextId);
+                    if (e.action.sound.distance == 1)
+                        target->PlayDistanceSound(e.action.sound.sound, e.action.sound.onlySelf ? target->ToPlayer() : nullptr, e.action.sound.keyBroadcastTextId);
```
Comment (Shauren) [inline src/server/game/AI/SmartScripts/SmartScript.cpp:301]: `PlayDistanceSound` doesn't have `keyBroadcastTextId` arg (no such field in that packet)

Comment (illfated) [reply src/server/game/AI/SmartScripts/SmartScript.cpp:301]: That argument seems to be a deviation from the originating (cherry-picked) commit https://github.com/TrinityCore/TrinityCore/commit/37995ade2991ed27845728eb6a6dd95f3a490849 , it is not present there.

Comment (funjoker) [reply src/server/game/AI/SmartScripts/SmartScript.cpp:301]: Will fix that

**Review by Shauren — APPROVED** [review]

**Pull request merged**